### PR TITLE
Added graphQL feature for Razorpay order

### PR DESCRIPTION
**Pull request by enishant**

### CODE REVIEW

File: Model/Resolver/setRzpPaymentDetailsOnCart.php [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+use Magento\QuoteGraphQl\Model\Cart\CheckCartCheckoutAllowance;
+use Magento\QuoteGraphQl\Model\Cart\GetCartForUser;
+use Magento\QuoteGraphQl\Model\Cart\SetPaymentMethodOnCart as SetPaymentMethodOnCartModel;
+use Razorpay\Magento\Model\PaymentMethod;
+use Razorpay\Magento\Model\Config;
+
+/**
+ * Mutation resolver for setting payment method for shopping cart
+ */
+class SetRzpPaymentDetailsOnCart implements ResolverInterface
```
Comment (ramth05) [inline Model/Resolver/setRzpPaymentDetailsOnCart.php]: @enishant , this need to be renamed `SetRzpPaymentDetailsForOrder`

File: Model/Resolver/setRzpPaymentDetailsOnCart.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        if (empty($args['input']['rzp_payment_id'])) {
+            throw new GraphQlInputException(__('Required parameter "rzp_payment_id" is missing.'));
+        }
+
+        $rzp_payment_id = $args['input']['rzp_payment_id'];
+        if (empty($args['input']['rzp_order_id'])) {
+            throw new GraphQlInputException(__('Required parameter "rzp_order_id" is missing.'));
+        }
+
+        $rzp_order_id = $args['input']['rzp_order_id'];
```
Comment (ramth05) [inline Model/Resolver/setRzpPaymentDetailsOnCart.php]: @enishant , we need to use this from our DB not directly from the user input.

File: Model/Resolver/PlaceRazorpayOrder.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                'currency'        => $salesOrder['order_currency_code'],
+                'payment_capture' => $payment_capture,
+                'app_offer'       => (($salesOrder['grand_total'] - $salesOrder['base_discount_amount']) > 0) ? 1 : 0,
+            ]);
+            if (null !== $order && !empty($order->id)) {
+
+                $responseContent = [
+                    'success'        => true,
+                    'rzp_order_id'   => $order->id,
+                    'order_quote_id' => $receipt_id,
```
Comment (ramth05) [inline Model/Resolver/PlaceRazorpayOrder.php]: @enishant , this should be `order_id`

File: Model/Resolver/PlaceRazorpayOrder.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    'amount'         => number_format((float) $salesOrder['grand_total'], 2, ".", ""),
+                    'currency'       => $salesOrder['order_currency_code'],
+                    'message'        => 'Razorpay Order created successfully'
+                ];
+                return $responseContent;
```
Comment (ramth05) [inline Model/Resolver/PlaceRazorpayOrder.php:146]: @enishant , we needs to store `rzp_order_id` on server.

File: Model/Resolver/PlaceRazorpayOrder.php (hunk 3 above)
Comment (enishant) [reply Model/Resolver/PlaceRazorpayOrder.php]: @ram updated order_quote_id to order_id

File: Model/Resolver/PlaceRazorpayOrder.php (hunk 4 above)
Comment (enishant) [reply Model/Resolver/PlaceRazorpayOrder.php:146]: @ramth05 stored rzp_order_id to database at table sales_order

File: Model/Resolver/setRzpPaymentDetailsOnCart.php (hunk 2 above)
Comment (enishant) [reply Model/Resolver/setRzpPaymentDetailsOnCart.php]: @ramth05 updated code to fetch rzp_order_id from database

File: Model/Resolver/setRzpPaymentDetailsForOrder.php [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        $this->scopeConfig                = $scopeConfig;
+        $this->_objectManager             = \Magento\Framework\App\ObjectManager::getInstance();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        if (empty($args['input']['order_id'])) {
```
Comment (ramth05) [inline Model/Resolver/setRzpPaymentDetailsForOrder.php]: @enishant , can we follow the `{` must start with new line every where in the code.

File: Model/Resolver/setRzpPaymentDetailsForOrder.php [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            throw new GraphQlInputException(__('Required parameter "order_id" is missing.'));
+        }
+
+        $order_id = $args['input']['order_id'];
```
Comment (ramth05) [inline Model/Resolver/setRzpPaymentDetailsForOrder.php]: @enishant , line gap required here.

File: Model/Resolver/setRzpPaymentDetailsForOrder.php [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        $rzp_signature = $args['input']['rzp_signature'];
+
+        $rzp_order_id = '';
+        try {
+            $collection     = $this->_objectManager->get(\Magento\Sales\Model\Order::class)
+            ->getCollection()
+            ->addFieldToSelect('*')
+            ->addFilter('increment_id', $order_id)
+            ->getFirstItem();
+            $salesOrder = $collection->getData();
```
Comment (ramth05) [inline Model/Resolver/setRzpPaymentDetailsForOrder.php]: @enishant , line gap. Please follow across the changes.

File: Model/Resolver/setRzpPaymentDetailsOnCart.php (hunk 1 above)
Comment (enishant) [reply Model/Resolver/setRzpPaymentDetailsOnCart.php]: @ramth05 Renamed resolver name SetRzpPaymentDetailsOnCart to setRzpPaymentDetailsForOrder

File: Model/Resolver/setRzpPaymentDetailsForOrder.php (hunk 5 above)
Comment (enishant) [reply Model/Resolver/setRzpPaymentDetailsForOrder.php]: @ramth05 Updated code to begin `{` on new line every where in the code.

File: Model/Resolver/setRzpPaymentDetailsForOrder.php (hunk 6 above)
Comment (enishant) [reply Model/Resolver/setRzpPaymentDetailsForOrder.php]: @ramth05 Added line gap across the changes

File: Model/Resolver/setRzpPaymentDetailsForOrder.php (hunk 7 above)
Comment (enishant) [reply Model/Resolver/setRzpPaymentDetailsForOrder.php]: @ramth05 Added line gap across the changes

File: Setup/UpgradeSchema.php [hunk 8]
```diff
@@ -38,6 +38,22 @@ public function upgrade(SchemaSetupInterface $setup,
...
+        if (version_compare($context->getVersion(), 'beta-3.3.4', '>')) {
+            $tableName = $setup->getTable('sales_order');
+            if ($setup->getConnection()->isTableExists($tableName) == true) {
+                $setup->getConnection()->addColumn(
+                    $tableName,
+                    'rzp_order_id',
+                    [
+                        'nullable' => true,
+                        'type'     => \Magento\Framework\DB\Ddl\Table::TYPE_TEXT,
+                        'length'   => 255,
```
Comment (ramth05) [inline Setup/UpgradeSchema.php]: @enishant , can we make length to just 55 here.

Comment (enishant) [reply Setup/UpgradeSchema.php]: @ramth05 updated length to 55

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+        }
+
+        $rzp_signature = $args['input']['rzp_signature'];
+
+        $rzp_order_id = '';
+        try
+        {
+            $collection     = $this->_objectManager->get(\Magento\Sales\Model\Order::class)
+            ->getCollection()
+            ->addFieldToSelect('*')
```
Comment (ramth05) [inline Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @enishant , why to select all the fields here?

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+            ->addFilter('increment_id', $order_id)
+            ->getFirstItem();
+
+            $salesOrder = $collection->getData();
+
+            $order = $this->order->load($salesOrder['entity_id']);
```
Comment (ramth05) [inline Model/Resolver/SetRzpPaymentDetailsForOrder.php]: We can fetch the the RZP order Id from above SQL only. why loading the object again.

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+
+declare(strict_types=1);
+
+namespace Razorpay\Magento\Model\Resolver;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\QuoteGraphQl\Model\Cart\CheckCartCheckoutAllowance;
```
Comment (ramth05) [inline Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @enishant , are we using these anywhere?

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php [hunk 12]
```diff
@@ -133,15 +124,7 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
...
-            $collection     = $this->_objectManager->get(\Magento\Sales\Model\Order::class)
-            ->getCollection()
-            ->addFieldToSelect('*')
-            ->addFilter('increment_id', $order_id)
-            ->getFirstItem();
-
-            $salesOrder = $collection->getData();
-
-            $order = $this->order->load($salesOrder['entity_id']);
+            $order = $this->order->load($order_id,$this->order::INCREMENT_ID);
```
Comment (ramth05) [inline Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @enishant space required after `,`

File: Model/Resolver/PlaceRazorpayOrder.php [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+        }
+        try
+        {
+            $storeScope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
+            $storeId = (int) $context->getExtensionAttributes()->getStore()->getId();
+            $receipt_id        = $args['order_id'];
+
+            $collection = $this->_objectManager->get(\Magento\Sales\Model\Order::class)
+            ->getCollection()
+            ->addFieldToSelect('*')
```
Comment (ramth05) [inline Model/Resolver/PlaceRazorpayOrder.php]: @enishant , can we apply similar changes here as well.

File: Model/Resolver/PlaceRazorpayOrder.php (hunk 4 above)
Comment (ramth05) [reply Model/Resolver/PlaceRazorpayOrder.php:146]: @enishant , looks like we are missing logger here as well.

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php (hunk 9 above)
Comment (enishant) [reply Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @ramth05 update query for order.

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php (hunk 10 above)
Comment (enishant) [reply Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @ramth05 update query for order.

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php (hunk 11 above)
Comment (enishant) [reply Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @ramth05 removed unused objects

File: Model/Resolver/SetRzpPaymentDetailsForOrder.php (hunk 12 above)
Comment (enishant) [reply Model/Resolver/SetRzpPaymentDetailsForOrder.php]: @ramth05 updated space after `,`

File: Model/Resolver/PlaceRazorpayOrder.php (hunk 13 above)
Comment (enishant) [reply Model/Resolver/PlaceRazorpayOrder.php]: @ramth05 Updated order query

File: Model/Resolver/PlaceRazorpayOrder.php (hunk 4 above)
Comment (enishant) [reply Model/Resolver/PlaceRazorpayOrder.php:146]: @ramth05 Added logger to PlaceRazorpayOrder & setRazorpayPaymentDetailsforOder

**Review by ramth05 — APPROVED** [review]
LGTM

**Pull request merged**